### PR TITLE
Add Ledger Connect Kit support

### PR DESCRIPTION
**Pull request by hlopes-ledger**
### Description

Replaces the old USB connection aproach with the new Connect Kit.

The Ledger Connect Kit loader gets the latest Connect Kit minor version from a CDN at runtime so that Ledger can release updates without waiting for wallet libraries or DApps to explicitly update packages and deploy new versions.

Depending on the user's platform, Connect Kit's UI will guide them to install the Ledger Connect extension in case it is supported, or present the required details to connect to Ledger Live through WalletConnect.

### Checklist
- [ ] The version field in `package.json` is incremented following [semantic versioning](https://semver.org/)
- [x] The box that allows repo maintainers to update this PR is checked
- [x] I tested locally to make sure this feature/fix works
- [x] I have run `yarn file-check`, `yarn type-check` & `yarn build` to confirm there are not any associated errors
- [ ] This PR passes the Circle CI checks

### CODE REVIEW

File: packages/ledger/src/index.ts [hunk 1]
```diff
@@ -1,434 +1,297 @@
...
+                  throw new ProviderRpcError({
+                    code: ProviderRpcErrorCode.INVALID_PARAMS,
+                    message: `The Provider requires a chainId to be passed in as an argument`
+                  })
+                }
 
-            let populatedTransaction = await signer.populateTransaction(
-              transactionObject
-            )
+                return this.connector.sendCustomRequest({
```
[the remark this answers is not in the document]
Comment (Adamj1232) [reply packages/ledger/src/index.ts:218]: Screen shot of modal with this method set as unsupported
Screen Shot 2023-01-04 at 3 16 32 PM

[the remark this answers is not in the document]
Comment (hlopes-ledger) [reply packages/ledger/src/index.ts:218]: Yes, I think the modal is OK until we implement WalletConnect v2 on Ledger Live.
How is the modal added?

**Comment by Adamj1232** [comment]
@hlopes-ledger Looks great and working well. Will update as WC-v2 evolves. One final question has you team across build issues with the new Ledger dep package? 
We currently have build documentation for Webpack 4/5, CRACO, React-Rewired, SvelteKit, Vite and Nuxt.js and can be added to the [Core package README here](https://github.com/blocknative/web3-onboard/blob/v2-web3-onboard-develop/packages/core/README.md) or the Ledger package README and will really help us troubleshooting in the future.
Thanks again!

**Comment by hlopes-ledger** [comment]
> @hlopes-ledger Looks great and working well. Will update as WC-v2 evolves. One final question has you team across build issues with the new Ledger dep package?
> We currently have build documentation for Webpack 4/5, CRACO, React-Rewired, SvelteKit, Vite and Nuxt.js and can be added to the [Core package README here](https://github.com/blocknative/web3-onboard/blob/v2-web3-onboard-develop/packages/core/README.md) or the Ledger package README and will really help us troubleshooting in the future.
> Thanks again!

@Adamj1232 We didn't. We're bundling the dependencies in Connect Kit itself and I've tested the connector in all the web3-onboard example apps on the repo and it works without any problems in next 12, next 13, vite-react and vue.

I couldn't get the svelte and vue-v2 examples to work (even without adding the Ledger connector). On the svelte one I get `Cannot mix different versions of joi schemas` on the console and a 500 error page on the browser. On the vue-v2 I get `Could not resolve "@vue/composition-api/dist/vue-composition-api.mjs"` and it doesn't start.

**Comment by Adamj1232** [comment]
@hlopes-ledger sounds good, we will reach out or troubleshoot and update the docs should any build issues arise. Will test on final time and get this released today. Thanks again for all your awesome work!

**Review by Adamj1232 — APPROVED** [review]
Nice work @hlopes-ledger !

**Pull request merged**